test/base/number.test.js

```diff
@@ -56,6 +39,128 @@ it.describe("comb/base/number.js",function (it) {
         });
+
+
+        it.should("roundHalfDown properly", function () {
+            assert.equal(comb.number.roundHalfDown(0.225, 2), 0.22);
+            assert.equal(comb.number.roundHalfDown(10.384, 2), 10.38);
+            assert.equal(comb.number.roundHalfDown(10.386, 2), 10.38);
+            assert.equal(comb.number.roundHalfDown(10.3869, 3), 10.386);
+            assert.equal(comb.number.roundHalfDown(10.3861, 3), 10.386);
+            assert.equal(comb.number.roundHalfDown(10.269019, 5), 10.26901);
+
+            assert.equal(comb(0.225).roundHalfDown(2), 0.22);
+            assert.equal(comb(10.384).roundHalfDown(2), 10.38);
+            assert.equal(comb(10.386).roundHalfDown(2), 10.38);
+            assert.equal(comb(10.3869).roundHalfDown(3), 10.386);
+            assert.equal(comb(10.3861).roundHalfDown(3), 10.386);
+            assert.equal(comb(10.269019).roundHalfDown(5), 10.26901);
+        });
+
+
+        it.should("roundHalfUp properly", function () {
+            assert.equal(comb.number.roundHalfUp(-7.6, 0), -8);
+            assert.equal(comb.number.roundHalfUp(-7.4, 0), -7);
+            assert.equal(comb.number.roundHalfUp(-3.515, 2), -3.51);
+            assert.equal(comb.number.roundHalfUp(-0.515, 1), -0.5);
+            assert.equal(comb.number.roundHalfUp(-0.515, 2), -0.51);
+            assert.equal(comb.number.roundHalfUp(-0.515, 3), -0.515);
+            assert.equal(comb.number.roundHalfUp(-0.5155, 4), -0.5155);
+            assert.equal(comb.number.roundHalfUp(-0.51551, 5), -0.51551);
+            assert.equal(comb.number.roundHalfUp(0.225, 2), 0.23);
+            assert.equal(comb.number.roundHalfUp(10.384, 2), 10.38);
+            assert.equal(comb.number.roundHalfUp(10.386, 2), 10.39);
+            assert.equal(comb.number.roundHalfUp(10.3869, 3), 10.387);
+            assert.equal(comb.number.roundHalfUp(10.3861, 3), 10.386);
+            assert.equal(comb.number.roundHalfUp(10.269019, 5), 10.26902);
+            assert.equal(comb.number.roundHalfUp(2.384, 2), 2.38);
+            assert.equal(comb.number.roundHalfUp(2.385, 2), 2.39);
+            assert.equal(comb.number.roundHalfUp(2.386, 2), 2.39);
+            assert.equal(comb.number.roundHalfUp(3.285151515, 8), 3.28515152);
+            assert.equal(comb.number.roundHalfUp(3.285151514, 8), 3.28515151);
+            assert.equal(comb.number.roundHalfUp(3.285151516, 8), 3.28515152);
+            assert.equal(comb.number.roundHalfUp(3.2851515155, 9), 3.285151516);
+            assert.equal(comb.number.roundHalfUp(3.2851515154, 9), 3.285151515);
+            assert.equal(comb.number.roundHalfUp(3.2851515156, 9), 3.285151516);
+            assert.equal(comb.number.roundHalfUp(3.28515151515, 10), 3.2851515152);
+            assert.equal(comb.number.roundHalfUp(3.28515151514, 10), 3.2851515151);
+            assert.equal(comb.number.roundHalfUp(3.28515151516, 10), 3.2851515152);
+            assert.equal(comb.number.roundHalfUp(0.985, 2), 0.99);
```

Comment:
Initially I was testing to make sure the same value ran multiple times would produce the same result. I'll clean them up though.